pkg/ast/spl/spl.peg

```diff
@@ -671,6 +690,185 @@ GroupbyBlock <- BY fields:FieldNameList {
+    }
+    ssOption := &structs.StreamStatsOptions{
+        TimeWindow: spanLen,
+    }
+    ssOptionArg := &StreamStatsOptionArgs{
+        argOption: "time_window",
+        streamStatsOptions: ssOption,
+    }
+    
+    return ssOptionArg, nil
+}
+
+StreamStatsOption <- ssOption:(StreamStatsOptionResetOnChange / StreamStatsOptionCurrent / StreamStatsOptionGlobal / StreamStatsOptionAllNum / StreamStatsOptionWindow / StreamStatsOptionResetBefore / StreamStatsOptionResetAfter / StreamStatsOptionTimeWindow) {
+    return ssOption, nil
+}
+
+StreamStatsOptionList <- first:StreamStatsOption rest:(SPACE StreamStatsOption)* {
+    restSlice := rest.([]any)
+    optionWasSpecified := make(map[string]bool)
+
+    numOptions := 1 + len(restSlice)
+
+    ssOption := initializeStreamStatsOptions()
+
+    for i := 0; i < numOptions; i++ {
+        var ssOptArg *StreamStatsOptionArgs
+        if i == 0 {
+            ssOptArg = first.(*StreamStatsOptionArgs)
+        } else {
+            separator := restSlice[i - 1].([]any)
+            ssOptArg = separator[1].(*StreamStatsOptionArgs)
+        }
+
+        argOption := ssOptArg.argOption
+        val, exist := optionWasSpecified[argOption]
+        if exist && val {
+            return nil, fmt.Errorf("Invalid Syntax, option: %v is already specified", argOption)
+        }
+        optionWasSpecified[argOption] = true
+
+        switch argOption {
+            case "reset_on_change":
+                ssOption.ResetOnChange = ssOptArg.streamStatsOptions.ResetOnChange
+            case "current":
+                ssOption.Current = ssOptArg.streamStatsOptions.Current
+            case "global":
+                ssOption.Global = ssOptArg.streamStatsOptions.Global
+            case "allnum":
+                ssOption.AllNum = ssOptArg.streamStatsOptions.AllNum
+            case "window":
+                ssOption.Window = ssOptArg.streamStatsOptions.Window
+            case "time_window":
+                ssOption.TimeWindow = ssOptArg.streamStatsOptions.TimeWindow
+            case "reset_before":
+                ssOption.ResetBefore = ssOptArg.streamStatsOptions.ResetBefore
+            case "reset_after":
+                ssOption.ResetAfter = ssOptArg.streamStatsOptions.ResetAfter
+            default:
+                return nil, fmt.Errorf("Not a Valid Streamstats Command Option: %v", argOption)
+        }
+    }
+
+    return ssOption, nil
+}
+
+StreamStatsBlock <- PIPE CMD_STREAMSTATS ssOptionList:(StreamStatsOptionList) SPACE aggs:(CommonAggregatorBlock) {
+    aggNode := aggs.(*structs.QueryAggregators)
+
+    aggNode.StreamStatsOptions = ssOptionList.(*structs.StreamStatsOptions)
+
+    return aggNode, nil
+}
+/ PIPE CMD_STREAMSTATS aggs:(CommonAggregatorBlock) {
+    aggNode := aggs.(*structs.QueryAggregators)
+
+    aggNode.StreamStatsOptions = initializeStreamStatsOptions()
+
+    return aggNode, nil
+}
```

Comment:
nit: do you need both of these sections, or can you delete 864-870 and change the `ssOptionList` on line 857 to be optional and then you just need that block?

Reply:
See https://github.com/siglens/siglens/pull/1269#discussion_r1678275562